### ISSUE TITLE
Zstack requirement to change sshd_config for PermitRootLogin to "yes" seems a security issue

### ISSUE THREAD

**Issue by bmullan** [zstackorg/zstack#25]
During my initial attempts to install zstack I eventually found the documentation reference:

           http://zstack.org/tutorials/flat-network-ui.html

where it says:
                                 = = = = = = = = = 
Configure root user
The KVM host will need root user credentials of SSH, to allow Ansible to install necessary packages and to give the KVM agent full control of the host. As this tutorial use a single machine for both ZStack management node and KVM host, you will need to configure credentials for the root user.

CentOS:
    sudo su
    passwd root

Ubuntu:
You need to also enable root user in SSHD configuration.
    1. sudo su
    2. passwd root
    3. edit /etc/ssh/sshd_config
    4. comment out 'PermitRootLogin without-password'
    5. add 'PermitRootLogin yes'
    6. restart SSH: 'service ssh restart'
                                 = = = = = = = = = 

I know its stated that doing the above is required for Ansible... but in my mind and from everything I've read anywhere this permitting "root" login via ssh is NOT considered a Security "best practice".

This also seems a very odd config requirement for an IaaS Platform to require for obvious reasons.

Isn't that the whole reason why  "sudo" exists ?

Is there no way to resolve this by leaving PermitRootLogin to deny "root" login but utilize SUDO instead?

Brian

**Comment by zxwing** [comment]
yeah. This version has not hardened everything with sudo. We will definitely fix this in future version.
